Source/WebSocket_Android.cpp

```diff
@@ -0,0 +1,89 @@
+#include "WebSocket_Base.h"
+#include <AndroidExtensions/Globals.h>
+#include <AndroidExtensions/JavaWrappers.h>
+
+using namespace android::global;
+using namespace android::net;
+using namespace java::lang;
+using namespace java::io;
+using namespace java::net;
+using namespace java::websocket;
+
+namespace UrlLib
+{
+    class WebSocket::WSImpl : public WSImplBase
+    {
+    public:
+        ~WSImpl()
+        {
+            delete m_webSocket;
+        }
+
+        void Open(std::string url,
+            std::function<void(void)> onopen,
+            std::function<void(void)> onclose,
+            std::function<void(std::string)> onmessage,
+            std::function<void(void)> onerror)
+        {
+            open_callback = onopen;
+            close_callback = onclose;
+            message_callback = onmessage;
+            error_callback = onerror;
+
+            m_url = url;
+            m_readyState = ReadyState::Connecting;
+            m_webSocket = new WebSocketClient(url, open_callback_stored, close_callback_stored, message_callback, error_callback);
+        }
+
+        void Send(std::string message)
+        {
+            if (!m_webSocket)
+            {
+                throw std::runtime_error{"WebSocket is not initialized"};
+            }
+
+            if (m_readyState == ReadyState::Closed ||
+                m_readyState == ReadyState::Closing ||
+                m_readyState == ReadyState::Connecting)
+            {
+                error_callback();
+                throw std::runtime_error{"WebSocket is not open"};
+            }
+
+            m_webSocket->Send(message);
+        }
+
+        void Close()
+        {
+            if (m_readyState == ReadyState::Closing ||
+                m_readyState == ReadyState::Closed)
+            {
+                error_callback();
+                throw std::runtime_error{"WebSocket is already Closing/Closed"};
+            }
+            m_readyState = ReadyState::Closing;
+            m_webSocket->Close();
+        }
+
+    private:     
+        WebSocketClient* m_webSocket;
```

Comment:
It doesn't seem like this needs to be pointer. If it really cannot be just a normal class, then either use std::optional or std::unique_ptr to avoid having to call new/delete explicitly.